both/access_control.js

```diff
@@ -231,101 +229,75 @@ if(Meteor.isServer) {
         'createdAt',
       ];
 
-      if(_.difference(fields, allowedFields).length > 0) {
+      return _onlyAllowedFields(fields, allowedFields);
+    },
+    fetch: [ 'competitionId' ],
+  });
+
+  RoundProgresses.allow({
+    update: function(userId, progress, fields, modifier) {
+      if(getCannotManageCompetitionReason(userId, progress.competitionId)) {
         return false;
       }
-      return true;
+      var allowedFields = [
+        'done',
+        'total',
+      ];
+      return _onlyAllowedFields(fields, allowedFields);
     },
-    fetch: [ 'competitionId' ],
   });
 
   Results.allow({
     update: function(userId, result, fields, modifier) {
-      var competition = Competitions.findOne({ _id: result.competitionId }, { fields: { _id: 1 } });
-      if(getCannotManageCompetitionReason(userId, competition._id)) {
+      if(getCannotManageCompetitionReason(userId, result.competitionId)) {
         return false;
       }
-
       var allowedFields = [
         'solves',
 
         'updatedAt',
         'createdAt',
       ];
-
-      if(_.difference(fields, allowedFields).length > 0) {
-        return false;
-      }
-      return true;
+      return _onlyAllowedFields(fields, allowedFields);
     },
     fetch: [ 'competitionId' ],
   });
 
-  var allowedOwnRegistrationFields = [
-    'userId',
-    'competitionId',
-    'uniqueName',
-    'registeredEvents',
-    'guestCount',
-    'comments',
-
-    'updatedAt',
-    'createdAt',
-  ];
   Registrations.allow({
     insert: function(userId, registration) {
-      if(!getCannotManageCompetitionReason(userId, registration.competitionId)) {
-        // If you're allowed to manage the competition, then you can
-        // add any registration.
-        return true;
-      }
-      if(getCannotRegisterReasons(registration.competitionId)) {
-        return false;
-      }
-      // can only edit entries with their user id
-      if(registration.userId != userId) {
-        return false;
-      }
-
-      if(_.difference(fields, allowedOwnRegistrationFields).length > 0) {
-        return false;
-      }
-      return true;
+      return Registrations.userIsAllowed(userId, registration);
```

Comment:
Just to clarify the DRY idea, it would look something like this:

Things.attachSchema({
  name: {
    type: String,
    label: "Thing name",
    **updateAllowed: true,**
  },
  gizmoId: {
    type: String,
    label: "Gizmo id",
    optional: true,
    **updateAllowed: false,**
  },
  listed: {
    type: Boolean,
    label: "Listed",
    **updateAllowed: 'ifAdmin',**
  },